parser.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) HashiCorp, Inc.
+
+package mql
```

Comment:
A package comment that explains the syntax and provides example expressions would be helpful for me as a reviewer.

Reply:
ty!